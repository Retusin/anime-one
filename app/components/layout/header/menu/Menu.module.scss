.menu {
  @apply hidden md:flex gap-6 lg:gap-10;
}
