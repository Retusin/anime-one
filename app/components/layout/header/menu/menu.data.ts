export interface MenuDataParams {
  title: string;
  link: string;
}

export const MenuData: MenuDataParams[] = [
  {
    title: 'Anime',
    link: '/anime'
  },
  {
    title: 'Manga',
    link: '/manga'
  },
  {
    title: 'Characters',
    link: '/characters'
  },
  {
    title: 'Random',
    link: '/random'
  }
];
