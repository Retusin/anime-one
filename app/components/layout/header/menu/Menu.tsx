import { FC } from 'react';

import { MenuItem } from './menu-item/MenuItem';
import { MenuData } from './menu.data';

import styles from './Menu.module.scss';

export const Menu: FC = () => (
  <div>
    <ul className={styles.menu}>
      {MenuData.map((item) => (
        <MenuItem item={item} key={item.title} />
      ))}
    </ul>
  </div>
);
