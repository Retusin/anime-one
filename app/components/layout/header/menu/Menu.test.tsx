import { Menu } from '@/components/layout/header/menu/Menu';
import { render } from '@testing-library/react';

test('render menu', () => {
  const { container } = render(<Menu />);
  // @ts-ignore
  expect(container).toMatchSnapshot();
});
