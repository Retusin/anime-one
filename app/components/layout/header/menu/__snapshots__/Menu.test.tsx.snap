// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render menu 1`] = `
<div>
  <div>
    <ul
      class="menu"
    >
      <li>
        <span
          class="menu_item"
        >
          Anime
        </span>
      </li>
      <li>
        <span
          class="menu_item"
        >
          Manga
        </span>
      </li>
      <li>
        <span
          class="menu_item"
        >
          Characters
        </span>
      </li>
      <li>
        <span
          class="menu_item"
        >
          Random
        </span>
      </li>
    </ul>
  </div>
</div>
`;
